nano/node/bootstrap/bootstrap.cpp

```diff
@@ -879,7 +883,10 @@ void nano::bootstrap_attempt::lazy_clear ()
 void nano::bootstrap_attempt::lazy_run ()
 {
 	assert (!node->flags.disable_lazy_bootstrap);
-	populate_connections ();
+	if (!populate_connections_started)
```

Comment:
I think we want `if (populate_connections_started.exchange(true)` everywhere, otherwise there's a race (and remove the assignment in populate_connections). Maybe factor that out into 

```
void try_populate_connections()
{
    if (populate_connections_started.exchange(true)
    {
            populate_connections ();
    }
}
```
?